tests/baselines/reference/controlFlowWhileStatement.js

```diff
@@ -105,6 +105,29 @@ function h3() {
     }
     x; // string | number
 }
+// Repro for #8418
+function foo(x: number): number { return 1; }
+function test1() {
+    let x: number | undefined;
+    while (cond) {
+        while (cond) {
+            while (cond) {
+                x = foo(x);
```

Comment:
I thought the usage of `foo(x)` was supposed to be a possibly undefined error here.